Source/WebCore/loader/FTPDirectoryParser.cpp

```diff
@@ -970,35 +958,29 @@ FTPEntryType parseOneFTPLine(const char* line, ListState& state, ListResult& res
                 // there is exactly 1 space between filename and previous token in all
                 // outputs except old Hellsoft
                 if (!isOldHellsoft)
-                    result.filename = tokens[tokmarker + 3] + toklen[tokmarker + 3] + 1;
+                    result.filename = std::span(tokens[tokmarker + 3] + toklen[tokmarker + 3] + 1, &(line[linelen]) - result.filename.data());
                 else
-                    result.filename = tokens[tokmarker + 4];
-
-                result.filenameLength = (&(line[linelen]))
-                    - (result.filename);
+                    result.filename = std::span(tokens[tokmarker + 4], &(line[linelen]) - result.filename.data());
 
-                if (result.type == FTPLinkEntry && result.filenameLength > 4) {
+                if (result.type == FTPLinkEntry && result.filename.size() > 4) {
                     /* First try to use result.fe_size to find " -> " sequence.
                        This can give proper result for cases like "aaa -> bbb -> ccc". */
                     auto fileSize = parseIntegerAllowingTrailingJunk<unsigned>(result.fileSize).value_or(0);
 
-                    if (result.filenameLength > (fileSize + 4) && !strncmp(result.filename + result.filenameLength - fileSize - 4, " -> ", 4)) {
-                        result.linkname = result.filename + (result.filenameLength - fileSize);
-                        result.linknameLength = (&(line[linelen])) - (result.linkname);
-                        result.filenameLength -= fileSize + 4;
+                    if (result.filename.size() > (fileSize + 4) && !strncmp(result.filename.data() + result.filename.size() - fileSize - 4, " -> ", 4)) {
```

Comment:
The `strncmp` call here doesn't seem super safe.

Reply:
Yes, this whole filed is currently marked as `WTF_ALLOW_UNSAFE_BUFFER_USAGE`. It will need quite a bit of work to make it buffer-safe, though I can look into this particular strncmp() call.